qa/os/src/test/java/org/elasticsearch/packaging/test/ArchiveTestCase.java

```diff
@@ -263,6 +263,64 @@ public void test60AutoCreateKeystore() throws Exception {
         });
     }
 
+    public void test61PasswordProtectedKeystore() throws Exception {
+        assumeThat(installation, is(notNullValue()));
+        // cleanup from previous test
+        // rm(installation.config("elasticsearch.keystore"));
+
+        String password = "keystorepass";
+        final Shell sh = new Shell();
+        Platforms.onLinux(() -> {
+            String systemJavaHome = sh.run("echo $SYSTEM_JAVA_HOME").stdout.trim();
+            sh.getEnv().put("JAVA_HOME", systemJavaHome);
```

Comment:
The `newShell()` method handles all this. Additionally, with https://github.com/elastic/elasticsearch/pull/45761 there is a protected `sh` variable already created for each test.